node/src/types/json_compatibility/public_key.rs

```diff
@@ -0,0 +1,15 @@
+use hex_fmt::HexFmt;
+use serde::{Deserialize, Serialize};
+
+/// Stringified PublicKey encoding.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct PublicKey(String);
+
+impl From<casper_types::PublicKey> for PublicKey {
+    fn from(public_key: casper_types::PublicKey) -> Self {
+        match public_key {
+            casper_types::PublicKey::Ed25519(inner) => PublicKey(format!("{:10}", HexFmt(inner))),
+            casper_types::PublicKey::Secp256k1(inner) => PublicKey(format!("{:10}", HexFmt(inner))),
```

Comment:
Will this abbreviated form be sufficient?  If a user wants to copy-paste the returned public key in order to enter it into a subsequent client command, they won't be able to do that.

That may be ok here though?

Reply:
There's no subsequent command that I'm aware of that would require this.